# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: ipc/glue/UtilityProcessChild.cpp
# Commit: cecf09ccbda3
# Full Hash: cecf09ccbda3f2ba21a49b13e6d8c83f61989cee
# Author: Ray Kraesig <rkraesig@mozilla.com>
# Date: 2023-11-08 21:12:03
# Description:
#   Bug 1863044 - remove unneeded variable and associated MOZ_ASSERT  r=ipc-reviewers,nika
#   
#   `mFileDialogInstance` is not needed, since we don't reuse old file-
#   dialog actors. (We _do_ reuse active file-dialog utility processes as
#   hosts for those actors; this check is a leftover from when that was
# ==============================================================================

diff -r b47bf0a3b904 -r cecf09ccbda3 ipc/glue/UtilityProcessChild.cpp
--- a/ipc/glue/UtilityProcessChild.cpp	Wed Nov 08 16:56:44 2023 +0000
+++ b/ipc/glue/UtilityProcessChild.cpp	Wed Nov 08 17:25:57 2023 +0000
@@ -310,9 +310,6 @@
       "UtilityProcessChild::RecvStartWinFileDialogService", OTHER,
       MarkerOptions(MarkerTiming::IntervalUntilNowFrom(mChildStartTime)));
 
-  MOZ_RELEASE_ASSERT(!mFileDialogInstance,
-                     "attempted to double-start file dialog service");
-
   auto instance = MakeRefPtr<widget::filedialog::WinFileDialogChild>();
   if (!instance) {
     return IPC_FAIL(this, "Failed to create WinFileDialogChild");
@@ -323,7 +320,6 @@
     return IPC_FAIL(this, "Failed to bind created WinFileDialogChild");
   }
 
-  mFileDialogInstance = std::move(instance);
   return IPC_OK();
 }
 